contrib/lib-autopilot.py

```diff
@@ -0,0 +1,289 @@
+
+        try:
+            self.__logger.info("Try to load graph from file system at: data/networkx_graph")
+            with open("data/networkx_graph","rb") as infile:
+                self.G = pickle.load(infile)
+                self.__logger.info("Successfully restored the lightning network graph from data/networkx_graph")
+        except FileNotFoundError:
+            self.__logger.info("load the graph from the peers of the lightning network")
+        
+            self.G = nx.Graph()
+            if self.__load_nodes()==False:
+                self.__logger.info("can not download nodes from the network and initialize the networkx Graph")
+            if self.__load_edges()==False:
+                self.__logger.info("cann not download the channels from the network and initialize the networkx Graph")
+            else:
+                with open("data/networkx_graph", "wb") as outfile:
+                    pickle.dump(self.G,outfile,pickle.HIGHEST_PROTOCOL)
+            
+    def __add_logger(self):
+        """ initiates the logging service for this class """
+        #FIXME: adapt to the settings that are proper for you
+        self.__logger = logging.getLogger('lib-autopilot')
+        self.__logger.setLevel(logging.INFO)
+        ch = logging.StreamHandler()
+        ch.setLevel(logging.INFO)
+        formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+        ch.setFormatter(formatter)
+        self.__logger.addHandler(ch)  
+        # TODO: CHANGE from Console to file handler
+        # fh = logging.FileHandler('lib-autopilot.log')
+        # fh.setLevel(logging.DEBUG)
+        #fh.setFormatter(formatter)
+        #self.__logger.addHandler(fh)
+        
+    def __load_nodes(self):
+        #FIXME: it is a real problem that we don't know how many nodes there could be
+        nodes = []
+        try:
+            self.__logger.info("Attempt RPC-call to download nodes from the lightning network")
+            while len(nodes) == 0: 
+                peers = self.__rpc_interface.listpeers()["peers"]
+                if len(peers) < 1:
+                    #FIXME: do we have seed nodes?
+                    self.__rpc_interface.connect("024a8228d764091fce2ed67e1a7404f83e38ea3c7cb42030a2789e73cf3b341365")
+                    time.sleep(2)
+                nodes = self.__rpc_interface.listnodes()["nodes"]
+        except ValueError as e:
+            self.__logger.info("Node list could not be retrieved from the peers of the lightning network")
+            self.__logger.debug("RPC error: " + str(e))
+            return False
+        self.__logger.info("Number of nodes found: {}".format(len(nodes)))
+
+        for node in nodes:
+            self.G.add_node(node["nodeid"],**node)
+        return True
+    
+    def __load_edges(self):
+        """
+        tries to store channels of the lightning network to the networkx graph. the attributes of the
```

Comment:
Nit: usually function comments should look like this:

```
"""Short one-line comment giving the synopsis

Much longer multiline comment explaining everything in minute details
"""
```

This integrates nicely with IDEs showing the short description and optionally expanding the long one.

Reply:
think I changed that everywhere